core/src/main/scala/kafka/server/FetchSession.scala

```diff
@@ -294,14 +294,24 @@ trait FetchContext extends Logging {
     */
   def updateAndGenerateResponseData(updates: FetchSession.RESP_MAP): FetchResponse
 
-  def partitionsToLogString(partitions: util.Collection[TopicPartition]): String =
-    FetchSession.partitionsToLogString(partitions, isTraceEnabled)
+  def partitionsToLogString(topics: FetchSession.RESP_MAP): String = {
```

Comment:
this method is used to log (DEBUG level) so it should be fine to iterate through whole collection.